e2e/tests/authflow/testcase.go

```diff
@@ -0,0 +1,168 @@
+package tests
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/authgear/authgear-server/pkg/lib/authflowclient"
+	"github.com/authgear/authgear-server/pkg/util/httputil"
+	"gopkg.in/yaml.v2"
+)
+
+func loadAllTestCases(path string) ([]TestCase, error) {
+	var testCases []TestCase
+	err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if filepath.Ext(path) != ".yaml" {
+			return nil
+		}
+
+		data, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		decoder := yaml.NewDecoder(strings.NewReader(string(data)))
+		for {
+			var testCase TestCase
+			err := decoder.Decode(&testCase)
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return err
+			}
+			testCase.Path = path
+			testCases = append(testCases, testCase)
+		}
+
+		return nil
+	})
+
+	return testCases, err
+}
+
+func runTestCase(t *testing.T, testCase TestCase) {
+	t.Logf("running test case: %s\n", testCase.Name)
+
+	ctx := context.Background()
+
+	appID := generateAppID()
+	e2eCmd := &End2EndCmd{
+		AppID:    appID,
+		TestCase: testCase,
+	}
+
+	err := e2eCmd.CreateConfigSource()
+	if err != nil {
+		t.Errorf("failed to create config source: %v", err)
+		return
+	}
+
+	for _, beforeHook := range testCase.Before {
+		switch beforeHook.Type {
+		case BeforeHookTypeUserImport:
+			err = e2eCmd.ImportUsers(beforeHook.UserImport)
+			if err != nil {
+				t.Errorf("failed to import users: %v", err)
+				return
+			}
+		case BeforeHookTypeCustomSQL:
+			err = e2eCmd.ExecuteCustomSQL(beforeHook.CustomSQL.Path)
+			if err != nil {
+				t.Errorf("failed to execute custom SQL: %v", err)
+				return
+			}
+		default:
+			t.Errorf("unknown before hook type: %s", beforeHook.Type)
+		}
+	}
+
+	hostName := httputil.HTTPHost(fmt.Sprintf("%s.portal.localhost:4000", appID))
+	client := authflowclient.NewClient(ctx, "localhost:4000", hostName)
```

Comment:
In my opinion, e2e tests should not use authflowclient. It should create a new client which call the authgear server like calling it from another project.

As authflowclient is not used except in e2e, I suggest rename it to e2e client and move to e2e package.